cmd/common/utils/parser-tableformatter.go

```diff
@@ -43,6 +47,7 @@ type GadgetParserOptions struct {
 // using the columns and formatter/textcolumns packages.
 type GadgetParser[T any] struct {
```

Comment:
Do you think `GadgetParser` is something we'll continue using in the future? I thought it was something temporal until the migration to use the columns library was completed on all gadgets, but now I see you're introducing new features to it. 

Is it something we really need? IIUC it's a wrapper/adapter around the columns library. I'm asking because I think many of the logic on this could go directly to the columns library, for instance, the `TransformToTable` function you implemented here.

Reply:
> Do you think `GadgetParser` is something we'll continue using in the future? 
>
I think we will. In particular, I like the idea of having this bridge that, on the one hand, avoids gadgets dealing with different packages: columns, textcolumns, sort, group, etc. And, on the other hand, it keeps those packages generic. Please notice that at some point, I expect to reach this:

```golang
func (p *GadgetParser[T]) Transform(entries ...*T) string {
	numEntries := len(entries)

	switch p.outputConfig.OutputMode {
	case OutputModeJSON:
		var b []byte
		var err error
		if numEntries == 1 {
			b, err = json.Marshal(entries[0])
		} else {
			b, err = json.MarshalIndent(entries, "", "  ")
		}

		if err != nil {
			fmt.Fprint(os.Stderr, fmt.Sprint(WrapInErrMarshalOutput(err)))
			return ""
		}

		return string(b)
	case OutputModeColumns:
		fallthrough
	case OutputModeCustomColumns:
		if numEntries == 1 {
			return p.formatter.FormatEntry(entries[0])
		} else {
			// Disable auto-scaling as AdjustWidthsToContent will already manage the
			// screen size.
			p.formatter.SetAutoScale(false)
			p.formatter.AdjustWidthsToContent(entries, true, textcolumns.GetTerminalWidth(), true)
			return p.formatter.FormatTable(entries)
		}
	default:
		fmt.Fprint(os.Stderr, fmt.Sprint(WrapInErrOutputModeNotSupported(p.outputConfig.OutputMode)))
		return ""
	}
}
```

> Is it something we really need? IIUC it's a wrapper/adapter around the columns library. I'm asking because I think many of the logic on this could go directly to the columns library, for instance, the TransformToTable function you implemented here.
>
As the previous example of a generic `Transform()`, I think the logic that involves the CLI flags can't be moved into the library. For instance, check my implementation proposal for adding `--k8s-metadata` flag. WDYT?

Reply:
Right. I think it's fine to have this small "glue" code there.